collections/pick.ts

```diff
@@ -28,5 +28,7 @@ export function pick<T extends object, K extends keyof T>(
   obj: Readonly<T>,
   keys: readonly K[],
 ): Pick<T, K> {
-  return Object.fromEntries(keys.map((k) => [k, obj[k]])) as Pick<T, K>;
+  return Object.fromEntries(
+    keys.flatMap((k) => (k in obj ? [[k, obj[k]]] : [])),
```

Comment:
I have not confirmed but using `flatMap` here would impact the performance a lot for a large object.